include/aspect/utilities.h

```diff
@@ -1232,6 +1232,29 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double SPD_safety_factor);
 
+    /**
+     * Computes the current_edot_ii and current_stress, with the option for
+     * viscous and viscoelastic contribution.
+     */
+    template <int dim>
+    double compute_current_edot_ii (const unsigned int j,  // the volume fraction
+                                    const std::vector<double> &composition,
+                                    const double ref_strain_rate,
+                                    const double min_strain_rate,
+                                    const SymmetricTensor<2,dim> &strain_rate,
+                                    const std::vector<double> &elastic_shear_moduli,
+                                    const bool use_elasticity,
+                                    const bool use_reference_strainrate);
+
+    /**
+     * Calculate the square root of the second moment invariant for the deviatoric
+     * strain rate tensor, including viscoelastic stresses.
+     */
+    double
```

Comment:
You are missing the `template <int dim>` before this line:
```suggestion
    template <int dim>
    double
```
This should make the error go away.

Reply:
thanks!